src/ui/PickColor.tsx

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from 'react'
+import { HexColorPicker } from 'react-colorful'
+import { setState, useStore } from '../store'
+
+export function PickColor(): JSX.Element {
+  const [color, setColor] = useState('#FFFF00')
+  const [pickcolor, set] = useStore((state) => [state.pickcolor, state.set])
+  const close = () => set((state) => ({ ...state, pickcolor: false }))
+
+  useEffect(() => {
```

Comment:
This seems like a hack.

Reply:
Which part? 
the **pickcolor** is the boolean to toggle the menu item, similar to the leaderboard.
the **close** is also similar to how the leaderboard menu is closed.
and the effect to detect the color changes. 

Please, let me know what part is weird so I can improve it.
Thanks for looking.

Reply:
I think you should be able to skip the local state and use the store directly.

``` ts
const setColor = (color) => set((state) => ({ ...state, color }))
```

Reply:
we have 2 variables here:
- color: the car color
- pickcolor: boolean to indicate whether the "pick color menu" is open or not

Same is been done here:
https://github.com/pmndrs/racing-game/blob/main/src/ui/LeaderBoard.tsx#L13